lib/opal/nodes/top.rb

```diff
@@ -108,6 +172,66 @@ def compile_end_construct
       def version_comment
         "/* Generated by Opal #{Opal::VERSION} */"
       end
+
+
+      def self.module_name(filenamepath, original_filepath = nil, original_filename = nil)
+        original_filename = filenamepath unless original_filename
+        original_filepath = Pathname.new(filenamepath).expand_path unless original_filepath
+        o_s = original_filepath.to_s
+        path, _ = Pathname.new(filenamepath).expand_path.split
+        if Opal.paths.include?(path.expand_path.to_s)
+          e = if o_s.end_with?('.js.rb')
+                -7
```

Comment:
magical constants? What's the intention of this code? To remove an extension suffix? `String#sub` with `/\.js\.rb\z/` can probably do it nicer. (Ruby 2.5 has a `String.delete_suffix` method, but we support older versions)